mod/search/start.php

```diff
@@ -524,3 +512,37 @@ function search_exclude_robots($hook, $type, $text) {
 
 	return $text;
 }
+
+/**
+ * Returns minimum and maximum lengths of words for MySQL search
+ * This function looks for stored config values, and, if none set,
+ * queries the DB and saves them
+ * @return stdClass An object with min and max properties
+ */
+function search_get_ft_min_max() {
+
+	$min = (int) elgg_get_config('search_ft_min_word_len');
+	$max = (int) elgg_get_config('search_ft_max_word_len');
+
+	if (!$min || !$max) {
```

Comment:
if you have min word length set to 0 this will always go... maybe not cast to int when fetching from config and check for null